File: src/main/java/ru/serdtsev/homemoney/moneyoper/RecurrenceOperRepo.java
package ru.serdtsev.homemoney.moneyoper;

import org.springframework.data.repository.CrudRepository;
import ru.serdtsev.homemoney.balancesheet.BalanceSheet;
import ru.serdtsev.homemoney.moneyoper.model.RecurrenceOper;

import java.util.UUID;
import java.util.stream.Stream;

public interface RecurrenceOperRepo extends CrudRepository<RecurrenceOper, UUID> {
  Stream<RecurrenceOper> findByBalanceSheet(BalanceSheet balanceSheet);
}

File: src/main/kotlin/ru/serdtsev/homemoney/moneyoper/RecurrenceOperRepo.kt
package ru.serdtsev.homemoney.moneyoper

import org.springframework.data.repository.CrudRepository
import ru.serdtsev.homemoney.balancesheet.BalanceSheet
import ru.serdtsev.homemoney.moneyoper.model.RecurrenceOper
import java.util.*

interface RecurrenceOperRepo : CrudRepository<RecurrenceOper, UUID> {
    fun findByBalanceSheet(balanceSheet: BalanceSheet): List<RecurrenceOper>
}